runtime/compiler/optimizer/SequentialStoreSimplifier.cpp

```diff
@@ -1707,7 +1708,7 @@ bool TR_arraycopySequentialStores::insertConsistentTree()
       }
 
    // make sure the value tree doesn't reference the address tree base var
-   if (_activeValueTree->getRootNode()->referencesSymbolInSubTree(baseRef, _comp->incOrResetVisitCount()))
+   if (_activeValueTree->getRootNode() == baseNode || _activeValueTree->getRootNode()->hasChild(baseNode))
```

Comment:
I wonder if we should leave this spot in the code unchanged. In the other places changed in this PR, node equality makes things more conservative, whereas I think in this place, it makes things less conservative than was the case before.

Reply:
I agree that it makes it less conservative, but thinking that it will be too conservative with the code not changed.
For field stores, any access of that field in the value tree (no matter the base array) would invalidate the transformation as symRefs will match.

Reply:
I think I see the issue, with node comparison if two access the same field and baseNode is different but might have the same base array value, this would allow it when it shouldn't.